viewing_party/party.py

```diff
@@ -1,23 +1,147 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
 
+    if title and genre and rating:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+        return movie
+    
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    if movie not in user_data["watched"]:
+        user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    if movie not in user_data["watchlist"]:
+        user_data["watchlist"].append(movie)
+
+    return user_data
+    
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            add_to_watched(user_data, movie)
+            user_data["watchlist"].remove(movie)
+
+    return user_data
+    
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0.0
+    
+    sum = 0.0
+
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    
+    avg_rating = sum / len(user_data["watched"])
+
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    
+    counts = {}
+    most_watched_genre = user_data["watched"][0]["genre"]
+
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+
+        if genre not in counts:
+            counts[genre] = 0
+            
+        counts[genre] += 1
+
+        if counts[most_watched_genre] < counts[genre]:
+            most_watched_genre = genre
+
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    friends_movies = get_friends_movies_list_no_duplicates(user_data["friends"])
+    unique_movies = get_unique_list_1_movies(user_data["watched"], friends_movies)
+
+    return unique_movies

```

Comment:
I love the way y'all have used your helper functions here to really streamline these particular functions (This one and the `get_friends_unique_watched`)!